components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -265,16 +232,18 @@ void AIChatTabHelper::OnInterceptedPageContentChanged() {
   }
 }
 
-// ai_chat::ConversationDriver
-
 GURL AIChatTabHelper::GetPageURL() const {
   return web_contents()->GetLastCommittedURL();
 }
 
-void AIChatTabHelper::GetPageContent(GetPageContentCallback callback,
-                                     std::string_view invalidation_token) {
-  if (IsPdf(web_contents()) && !is_pdf_a11y_info_loaded_) {
+void AIChatTabHelper::GetPageContent(
+    ConversationHandler::GetPageContentCallback callback,
+    std::string_view invalidation_token) {
+  bool is_pdf = IsPdf(web_contents());
+  if (is_pdf && !is_pdf_a11y_info_loaded_) {
     if (pending_get_page_content_callback_) {
+      // TODO(petemill): Queue the callback in a Signal, instead of only
```

Comment:
Same TODO as above, really, although this one more specific to a functional problem if called multiple times, instead of an optimization issue at the above TODO's location.

> It doesn't need to IMO because that is actually handled at the higher level (AssociatedContentDriver). However, this class doesn't know that. For non-pdf page's, GetPageContent can handle multiple calls at the same time (though it doesn't queue them, they'll just run independently). But for PDF documents, the existing pending_get_page_content_callback_ will get called with no content and only the latest call will be used. But it's not an issue with the way GetPageContent is called right now. So, the TODO is there as a direction in case AIChatTabHelper::GetPageContent gets used in other ways in the future.

...but if you mean should I have said "OneShotEvent" and not "Signal", because `Signal()` is what you call on a `OneShotEvent` - then yes! 😄